Libraries/ReactNative/AppContainer.js

```diff
@@ -46,17 +48,23 @@ class AppContainer extends React.Component<Props, State> {
 
   static getDerivedStateFromError: any = undefined;
 
-  static childContextTypes:
-    | any
-    | {|rootTag: React$PropType$Primitive<number>|} = {
-    rootTag: PropTypes.number,
-  };
+  static Context: React$Context<ContextType> = AppContainerContext;
 
-  getChildContext(): Context {
-    return {
-      rootTag: this.props.rootTag,
-    };
-  }
+  static childContextTypes:
+     | any		
+     | {|rootTag: React$PropType$Primitive<number>|} = {		
+     rootTag: PropTypes.number,		
+   };		
+
+    getChildContext(): ContextType {
+     console.warn(
+        'AppContainer has been migrated to the new Context API. ' +
+        'It is recommended to use AppContainer.Context.Consumer to consume the context.'
+      );
+     return {		
```

Comment:
What if you make this an object with a getter for rootTag that includes the warning. So the warning would hopefully only fire when the rootTag is read, not on every call to this function.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get